src/casanovoutils/downsample_spectra.py

```
@@ -0,0 +1,182 @@
+    if downsample_type not in _VALID_TYPES:
+        raise ValueError(
+            f"--downsample-type must be one of {sorted(_VALID_TYPES)}, "
+            f"got {downsample_type!r}."
+        )
+
+    if downsample_type == "number":
+        if downsample_rate != int(downsample_rate) or int(downsample_rate) < 1:
+            raise ValueError(
+                "--downsample-rate must be a positive integer when "
+                f"--downsample-type is 'number', got {downsample_rate!r}."
+            )
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)